Source/hydro/flatten.cpp

```diff
@@ -0,0 +1,166 @@
+#include "Castro.H"
+#include "Castro_F.H"
+#include "Castro_hydro_F.H"
+#include <cmath>
+
+#ifdef RADIATION
+#include "Radiation.H"
+#endif
+
+using namespace amrex;
+
+void
+Castro::uflatten(const Box& bx,
+                 Array4<Real const> const q,
+                 Array4<Real> const flatn, const int pres_comp) {
+
+  constexpr Real small_pres = 1.e-200;
+
+  // Knobs for detection of strong shock
+  constexpr Real shktst = 0.33;
+  constexpr Real zcut1 = 0.75;
+  constexpr Real zcut2 = 0.85;
+  constexpr Real dzcut = 1.0/(zcut2-zcut1);
+
+  AMREX_PARALLEL_FOR_3D(bx, i, j, k,
+  {
+
+    // x-direction flattening coef
+
+    Real dp = q(i+1,j,k,pres_comp) - q(i-1,j,k,pres_comp);
+
+    int ishft = dp > 0.0 ? 1 : -1;
+
+    Real denom = std::max(small_pres, std::abs(q(i+2,j,k,pres_comp) - q(i-2,j,k,pres_comp)));
+    Real zeta = std::abs(dp) / denom;
+    Real z = std::min(1.0, std::max(0.0, dzcut * (zeta - zcut1)));
```

Comment:
std::min and std::max are unavailable in CUDA device code. Please use amrex::min and amrex::max (they're defined in AMReX_Algorithm.H, and should be readily available here).